frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx

```diff
@@ -3,97 +3,138 @@ import * as _ from 'lodash';
+          const ports = getPorts(isi);
           setFieldValue('isSearchingForImage', false);
-        });
-    },
-    [setFieldValue, values.application.name, values.name, values.project.name],
-  );
+          setFieldValue('isi.name', name);
+          setFieldValue('isi.image', image);
+          setFieldValue('isi.tag', tag);
+          setFieldValue('isi.status', status);
+          setFieldValue('isi.ports', ports);
+          setFieldValue('image.ports', ports);
+          setFieldValue('image.tag', tag);
+          !values.name && setFieldValue('name', getSuggestedName(name));
+          !values.application.name &&
+            setFieldValue('application.name', `${getSuggestedName(name)}-app`);
+          // set default port value
+          const targetPort = _.head(ports);
+          targetPort && setFieldValue('route.targetPort', makePortName(targetPort));
+          setValidated(ValidatedOptions.success);
+        } else {
+          setFieldValue('isSearchingForImage', false);
+          setFieldValue('isi', {});
+          setFieldValue('isi.status', status.message);
+          setFieldValue('route.targetPort', null);
+          setValidated(ValidatedOptions.error);
+        }
+      })
+      .catch((error) => {
+        setFieldValue('isi', {});
+        setFieldValue('isi.status', error.message);
+        setFieldValue('isSearchingForImage', false);
+        setValidated(ValidatedOptions.error);
+      });
+  }, [setFieldValue, values.application.name, values.name, values.project.name, values.searchTerm]);
 
   const handleSave = (name: string) => {
     setNewImageSecret(name);
-    values.searchTerm && handleSearch(values.searchTerm);
+    values.searchTerm && handleSearch();
   };
 
+  const getHelpText = () => {
+    if (values.isSearchingForImage) {
+      return (
+        <span style={{ fontWeight: 'bold' }}>
+          <LoadingInline />
+        </span>
+      );
+    }
+    if (!values.isSearchingForImage && validated === ValidatedOptions.success) {
+      return (
+        <span style={{ fontWeight: 'bold', color: 'var(--pf-global--success-color--200)' }}>
+          <CheckCircleIcon /> Validated
+        </span>
+      );
+    }
+
+    return '';
+  };
+
+  const helpTextInvalid = validated === ValidatedOptions.error && (
+    <span className="odc-imagesearch-helptextinvalid">
+      <ExclamationCircleIcon />
+      &nbsp;
+      {values.searchTerm === '' ? 'Required' : values.isi.status}
+    </span>
+  );
+
   React.useEffect(() => {
-    !dirty && values.searchTerm && handleSearch(values.searchTerm);
+    !dirty && values.searchTerm && handleSearch();
   }, [dirty, handleSearch, values.searchTerm]);
 
   return (
     <>
-      <InputSearchField
+      <InputField
         type={TextInputTypes.text}
-        data-test-id="deploy-image-search-term"
         name="searchTerm"
-        label="Image Name"
-        onSearch={handleSearch}
+        placeholder="Enter an image name"
```

Comment:
@sahil143  seems the removal of `data-test-id="deploy-image-search-term"` willl impact integration tests

<img width="1431" alt="Screenshot 2020-02-13 at 5 09 02 PM" src="https://user-images.githubusercontent.com/5129024/74430850-b4400c00-4e83-11ea-8ca1-275e604fa32b.png">

Reply:
@invincibleJai added data-test-id

Reply:
Good catch @invincibleJai - we need to be careful not to remove `data-test-id`s.